packages/plugin-ext/src/main/browser/plugin-ext-frontend-module.ts

```diff
@@ -60,9 +60,26 @@ import { ViewColumnService } from './view-column-service';
 import { ViewContextKeyService } from './view/view-context-key-service';
 import { PluginViewWidget, PluginViewWidgetIdentifier } from './view/plugin-view-widget';
 import { TreeViewWidgetIdentifier, VIEW_ITEM_CONTEXT_MENU, PluginTree, TreeViewWidget, PluginTreeModel } from './view/tree-view-widget';
+import { RPCProtocol } from '../../common/rpc-protocol';
+import { LanguagesMainFactory, OutputChannelRegistryFactory } from '../../common';
+import { LanguagesMainImpl } from './languages-main';
+import { OutputChannelRegistryMainImpl } from './output-channel-registry-main';
 
 export default new ContainerModule((bind, unbind, isBound, rebind) => {
 
+    bind(LanguagesMainImpl).toSelf().inSingletonScope();
+    bind(LanguagesMainFactory).toFactory(context => (rpc: RPCProtocol) => {
+        const child = context.container.createChild();
+        child.bind(RPCProtocol).toConstantValue(rpc);
+        return child.get(LanguagesMainImpl);
+    });
+
+    bind(OutputChannelRegistryMainImpl).toSelf().inSingletonScope();
```

Comment:
not `inSingletonScope`, it has to be `inTransientScope`, otherwise we provide the same instance for all rpc, the same for `LanguagesMainImpl`

Reply:
`bind(OutputChannelRegistryFactory)` and `bind(LanguagesMainFactory)` has to be `inSingletonScope`

Reply:
@akosyakov 

> bind(OutputChannelRegistryFactory) and bind(LanguagesMainFactory) has to be inSingletonScope

Do you mean something like
```
bind(OutputChannelRegistryFactory).toFactory(context => () => {
        const child = context.container.createChild();
        return child.get(OutputChannelRegistryMainImpl);
}).inSingletonScope();
```

because it generates an error. 

I haven't yet found a way to add a factory to a singleton scope and there doesn't seem to be anything in the inversify documentation for it

Reply:
yes, factories are apparently singletons by default